lib/document.js

```diff
@@ -698,6 +698,19 @@ Document.prototype.markModified = function (path) {
 }
 
 /**
+ * Marks the path as not having changes to write to the db.
+ *
+ * _Helpful to prevent changes to certain fields based on logic in  middleware_
+ *
+ * @param {String} path the path to mark as not modified
+ * @api public
+ */
+
+Document.prototype.markNotModified = function (path) {
```

Comment:
Nope. The general idea would work, but this particular implementation won't work with modern mongoose.